crates/full-node/sov-sequencer/src/preferred/inner.rs

```diff
@@ -429,16 +428,30 @@ where
             unreachable!();
         }
 
+        let recovery_strategy = self
+            .seq_config
+            .sequencer_kind_config
+            .recovery_strategy
+            .clone();
+
         self.is_ready = Err(SequencerNotReadyDetails::PreferredSequencerRecovering);
         let next_sequence_number_according_to_node =
             get_next_sequence_number_according_to_node(info, &mut Rt::default());
+
         self.executor_events_sender
             .trigger_recovery(next_sequence_number_according_to_node, recovery_strategy)
             .await;
 
-        let executor_from_info = RollupBlockExecutor::<_, Rt>::new(info, rollup_exec_config);
+        // Creates a new executor  for recovery. This must *not* be called to create executors
+        // under other circumstances, since it causes side effects on the transaction cache.
```

Comment:
Since this is no longer a doc comment on a function:
```suggestion
        // Creates a new executor for recovery, which will cause side effects on the transaction cache.
```